lib/blocks.js

```diff
@@ -80,17 +127,83 @@ BlockController.prototype.show = function(req, res) {
   }
 };
 
-BlockController.prototype.blockIndex = function(req, res, next, height) {
-  var info = this.node.services.bitcoind.getBlockIndex(parseInt(height));
-  if(!info) {
-    return common.handleErrors(null, res);
+BlockController.prototype.showRaw = function(req, res) {
+  if (req.rawBlock) {
+    res.jsonp(req.rawBlock);
   }
+};
 
-  res.jsonp({
-    blockHash: info.hash
+BlockController.prototype.blockIndex = function(req, res, next, height) {
+  this.node.services.bitcoind.getBlockHeader(parseInt(height), function(err, info) {
+    if (err) {
+      return common.handleErrors(err, res);
+    }
+    res.jsonp({
+      blockHash: info.hash
+    });
   });
 };
 
+BlockController.prototype._getBlockSummary = function(hash, moreTs, next) {
+  var self = this;
+
+  function finish(result) {
+    if (moreTs > result.time) {
+      moreTs = result.time;
+    }
+    return next(null, result);
+  }
+
+  var summaryCache = self.blockSummaryCache.get(hash);
+
+  if (summaryCache) {
+    finish(summaryCache);
+  } else {
+    self.node.services.bitcoind.getRawBlock(hash, function(err, blockBuffer) {
+      if (err) {
+        return next(err);
+      }
+
+      var br = new bitcore.encoding.BufferReader(blockBuffer);
+
+      // take a shortcut to get number of transactions and the blocksize.
+      // Also reads the coinbase transaction and only that.
+      // Old code parsed all transactions in every block _and_ then encoded
+      // them all back together to get the binary size of the block.
+      // FIXME: This code might still read the whole block. Fixing that
```

Comment:
It depends if it's necessary to relieve performances issues. The caching has done a decent job for any recent summaries.